### ISSUE TITLE
[FEATURE REQUEST] Previews for CR Multi-ControlNet Stack

### ISSUE THREAD

**Issue by alessandroperilli** [Suzie1/ComfyUI_Comfyroll_CustomNodes#53]
The `CR Multi-ControlNet Stack` node is incredibly useful and I have it as a key feature of the AP Workflow.

One thing that would make it even better is if the node had 3 extra `image` outputs, each one generating a preview of the ControlNet effect, like it happens in A1111/SD Next:

![image-105](https://github.com/RockOfFire/ComfyUI_Comfyroll_CustomNodes/assets/9775949/fd2e19cb-885e-46c3-ac2e-4a602089a22a)

![image-110](https://github.com/RockOfFire/ComfyUI_Comfyroll_CustomNodes/assets/9775949/085b758b-9811-43af-a89c-ae85477ae54c)

Thank you for considering the request.

**Comment by Suzie1** [comment]
yes, i really want to add this too
the node is in my rework queue and will get be getting my full attention again soon
Suzie

**Comment by Suzie1** [comment]
Hi,
I am looking at this request today and would like to get your feedback.

First my understanding is that A1111 is simply displaying the output from the preprocesser selected. In ComfyUI this can be done by adding a preview node on the preprocessor like in the last image here:

https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/wiki/ControlNet-Nodes

I can add a 'preview' output on the Stack node that displays each of the input images. This could replace previews attached to the preprocessors. It wouldn't add much value other than allowing the 1, 2 or 3 preprocessed images to be displayed in one Preview Image node. I was thinking if adding text annotation would give this more value, but the preprocessor output is only an image, so the name of the preprocessor and the parameters would not be available to add as annotations.

After the models are applied in the Apply node, the result is conditioning which you cannot display a preview for.

I'm not sure if that is what you are looking for. Can you maybe clarify how you see it working?

**Comment by alessandroperilli** [comment]
Yes: given that we now have dozens of ControlNet models, it's hard to remember what each one does, especially for newcomers, and it's hard to tell if, say, Canny would do a better job than LineArt with a certain uploaded image. Seeing a preview of each preprocessor would help make an informed choice in the `CR Multi-ControlNet Stack `node. 

However, since I opened this PR, I realised that 6 previews wouldn't be enough for what I was trying to accomplish, so I created a DIY solution:

<img width="1641" alt="Screenshot 2023-12-11 at 13 40 56" src="https://github.com/Suzie1/ComfyUI_Comfyroll_CustomNodes/assets/9775949/b73dfc02-d530-4440-ab76-76f02f96687d">

The group on the left is bypassable via a controller in another part of the workflow. I enable it when I need to see which ControlNet models are going to be the best for any particular job, then I set the `CR Multi-ControlNet Stack `node in the group on the right accordingly, and then I disable the group on the left for the actual job.

I think this approach serves the intended purpose well, and it will be part of the upcoming AP Workflow 7.0. 

My original request was meant to eliminate the need to deploy `Preprocessor` nodes and too many `Preview Image` nodes, that's all.

Thanks for looking into this.
